internal/website/docs/next/reference/hooks/usePreviewNode.mdx

```diff
@@ -0,0 +1,82 @@
+---
+slug: /next/reference/hooks/usePreviewNode
+title: usePreviewNode Hook
+description: The usePreviewNode hook provides a way to get post preview data from WordPress for any post type
+---
+
+The `usePreviewNode` hook provides an abstraction around getting preview data for any post type. When calling the `usePreviewNode` hook on an appropriate preview page, the hook will ensure authorization, determine the post type, and make a request to retrieve the proper preview data.
+
+The following example shows how to use the `usePreviewNode` hook to render the default `post`/`page` post types:
+
+```tsx title="src/pages/preview.tsx"
+import { PageComponent } from 'pages/[...pageUri]';
+import { PostComponent } from 'pages/posts/[postSlug]';
+import { client, Page, Post } from 'client';
+
+export default function Preview() {
+  const isLoading = client.useIsLoading();
+  const { postType, node } = client.auth.usePreviewNode();
+
+  if (isLoading || node === undefined) {
+    return <p>loading...</p>;
+  }
+
+  if (node === null) {
+    return <p>Post not found</p>;
+  }
+
+  switch (postType) {
+    case 'Page': {
+      return <PageComponent page={node as Page} />;
+    }
+    case 'Post': {
+      return <PostComponent post={node as Post} />;
+    }
+    default: {
+      throw new Error(`Unspecified post type preview: ${postType}`);
+    }
+  }
+}
+```
+
+Additionally, the `usePreviewNode` hook can be used to get preview data for any custom post type. Say we have a custom post type called `Event`:
+
+```tsx title="src/pages/preview.tsx" {24-32}
+import { PageComponent } from 'pages/[...pageUri]';
+import { PostComponent } from 'pages/posts/[postSlug]';
+import { client, Page, Post, Event } from 'client';
+
+export default function Preview() {
+  const isLoading = client.useIsLoading();
+  const { postType, node } = client.auth.usePreviewNode();
+
+  if (isLoading || node === undefined) {
+    return <p>loading...</p>;
+  }
+
+  if (node === null) {
+    return <p>Post not found</p>;
+  }
+
+  switch (postType) {
+    case 'Page': {
+      return <PageComponent page={node as Page} />;
+    }
+    case 'Post': {
+      return <PostComponent post={node as Post} />;
+    }
+    case 'Event': {
+      const event = node as Event;
+
+      return (
```

Comment:
I noticed that if we are using a switch statement it triggers two Graphql requests. One for getting the __typename and one for retrieving the title and content. For example I have a CUSTOM POST type called Movie:

```
 switch (postType) {
    case 'Page': {
      return <PageComponent page={node as Page}/>;
    }
    case 'Post': {
      return <PostComponent post={node as Post}/>;
    }
    case 'Movie': {
      const n = node as Movie;
      return <div>{n.isPreview}</div>
    }
    // Add additional custom post types here as needed
    default: {
      throw new Error(`Unspecified post type preview: ${postType}`);
    }
```

Then I see the requests in the console:
<img width="1536" alt="Screenshot 2021-12-09 at 16 41 47" src="https://user-images.githubusercontent.com/328805/145439066-dedad1c4-6bf3-454a-8e0c-d28a6ea4752a.png">

<img width="1557" alt="Screenshot 2021-12-09 at 16 41 41" src="https://user-images.githubusercontent.com/328805/145439053-967fb5c6-2cbb-4780-99aa-94d80638f27f.png">

Is it possible to provide a way of avoid that? If not we should provide relevant documentation so that the users are aware of this behavior.

Reply:
@theodesp Yes, this is correct.

Something I had in mind we could do is add the `postType` in the preview URL so we wouldn't have to do this second request:

https://github.com/wpengine/faustjs/blob/90ad8b7eaf1edebe11055b8114edb6dbeae8c218/plugins/faustwp/includes/replacement/callbacks.php#L139